### ISSUE TITLE
The Scourge

### ISSUE THREAD

**Issue by zumbak04** [Warcraft-GoA-Development-Team/Warcraft-Guardians-of-Azeroth-2#44]
**Describe your suggestion in full detail below:**
The Scourge should be added as a gameplay feature.
- [ ] #350 
- [ ] #355

# Roadmap:
- [x] Add the **_In Ice Prison_** trait that blocks leading armies and tells Ner'zhul about his goal
  - [x] Draw icon, use this background:

<details>
<summary>Click to expand</summary>

![_frame_fame_bad](https://user-images.githubusercontent.com/46384992/135768371-517dd58a-fc10-4d3d-94b6-2ebeccc74407.png)

</details>

- [x] Add the **_Lich King_** with big prowess buff
  - [x] Draw icon, use this background:

<details>
<summary>Click to expand</summary>

![_frame_fame_good](https://user-images.githubusercontent.com/46384992/135768352-5323bbdb-96c8-4ee9-8a3f-21aeacca6e71.png)

</details>

- [x] After Draenor dies, the Scourge spawns with an undead army in Icecrown, Northrend rulers get notification
  - [x] Need text for notification (600-550 characters)
- [x] Scourge CB turns rulers undead
- [x] Vrykul should become vassals on their own
- [x] Undead under the Lich King shouldn't be able to reject the Lich King
- [ ] Story cycle for Ner'zhul:
  - [x] Spawns dreadlords to watch over Ner'zhul
    - [x] Event: If Ner'zhul imprisons or kills openly one of the dreadlords, another dreadlord executes Ner'zhul shattering the Scourge
      - [ ] Need text for this event (600-550 characters, from Ner'zhul perspective, second person)
  - [x] Ner'zhul has 15 years to summon the Legion
    - [x] Event: If he fails, a dreadlord kills him shattering the Scourge
      - [ ] Need text for this event (600-550 characters, from Ner'zhul perspective, second person)
    - [x] Event: When 10 years pass, dreadlords visit Ner'zhul reminding him he has 5 years
      - [ ] Need text for this event (600-550 characters, from Ner'zhul perspective, second person)
  - [x] Ner'zhul can't convert demons
  - [x] Event: If Ner'zhul kills all the dreadlords, the Scourge becomes free from the Legion
    - [x] Need text for this event (600-550 characters, from Ner'zhul perspective, second person)
  - [x] Forces AI Ner'zhul to expand in Northrend, Lordaeron, Quel'thalas, and Dalaran
  - [ ] When Ner'zhul summons the Legion, he may either die and return to his wife or lead the Scourge against the Legion
  - [x] Add special doctrine to Mord'truari as long as they are puppets of the Legion
- [x] Ner'zhul should spawn as a religious head
- [ ] Add a way to recreate the Scourge if it's destroyed

**Comment by SolTsuki** [comment]
I have two models to get sort of the same, one is way more simple but it doesn't have as much versatility as the other model, the second model... well, it's overkill:

---
I HAVE TO CHECK THIS DOCUMENT, IT'S PROBABLY UNREADABLE
---

- Simplified proposal
    - For the Lich King, you have a decision which is "Spread the Plague"
        - Choose enemy county
    - Then an event chain starts, like when you do a seduce scheme
        - it has a discover and a success chance, which change over time (every month)
        - sometimes you get events related to all of this like
            - you recruit cultists, with some intrigue and diplomacy options
            - you put the plague into the wheat
            - someone finds you, diplomacy and prowess options
            ...
        - if you get discovered you start having events that decrements your chances of success
        - if you fail, nothing happens, if you succeed
            - the county is plagued, which means
                - no levies, taxes for the owning player
                - the lich king player get special troops related to the infected place (levies will be a % of original levies of place for instance)
        - if place gets plagued, same chain event will happens elsewhere close, so the plague continues
        - when a county is plagued, owning player or lieges of owning player can "purge" the place
            - if owning player doesn't accept
                - some war similar to an independence war starts, if liege wins then short
                if liege lose then nothing happens
            - if owning player accepts, then you get the long chain

            - long chain - another chain event like the seduce one
                - initial events
                    - your troops find trouble through the way, strategy, prowess
                    - your troops actually don't wanna do a purge, strategy
                        - if goes wrong you lose part of the troops
                - middle events
                    - you find enemy forces, strategy, prowess
                    - you find zombified locals, strategy, prowess
                    - your own troops become infected, learning(?)
                - final events
                    - you find the one responsible, prowess
                    - you find a plague cauldron, learning
            - short chain
                - middle events like above
                - final events like above
            - if you manage to succeed on the chains, the place is no longer plagued
            - if the player owning the plagued place, dies, is converted to undead... then the place becomes
            part of the lich king domain, and domain is no longer plagued
    - how the Lordaeron plague would happen
        - LichKing makes decision to release plague on Brill (I'm assuming Warcraft 3 Brill's location)
        - event chain starts
            - cultists from the surrounding areas join (medium increase in success , also minor increase in discovery)
            - plague is introduced in wheat (high increase in success, minor increase in discovery)
            - Abomination created (medium increase in success, high increase in discovery)
            - Brill gets plagued
        - new event chain in Andorhal
            - some meatwagons are created (high increase in success, high increase in discovery)
            - abomination is created
            - plot discovered
            - plague lord gets killed (high decrease in sucess)
            - Andorhal is plagued
        - new event chain in Hearthglen
            - plagued provisions from previous place came (high increase suscess, medium increase in discovery
            - lich joins (high increase sucsess, low increase in discovery)
            - lich joins
            - plot gets discovered
            - sickened villagers killed by paladins (low decrease in success
            - order of paladins attack (high decrease in success)
            - Hearthglen is plagued
        - Stratholme happens
            - dreadlord joins (high increase in success, medium DECREASE in discovery)
            - abomation is created
            - all inhabitants eat plagued wheat (high increase, low discovery)
            - Stratholme is plagued
        - Lordaeron starts purge on Stratholme
            - in this case Stratholme is cool with being purged
            - starts event chain
                - troops are already close by (medium increase in success)
                - some troops don't want purge, you try a martial speech and fails (high decrease in success)
                - you have a paladin with the power of the Light on his side (medium increase in success)
                - you kill infected humans (medium increas in success)
                - Stratholme gets purged
        - from now one it's about purging and plaguing, also because you have a casus belli against the Lich King
        you can do a "holy war", with your sons Arthas as commander.

- (way more) Advanced (and complex) proposal
    - questions
        - are we gonna have a magic system? with characters being able to cast spells
            - create plague
            - mind control
            - portal - Mal'ganis has some dimensional powers to move more freely
            - basic necromancy
                - raise dead (active and passive, passive would be to that levies and champions on neighbouring counties
                will raise as undead)
        - purge
            - purge should only happen when a place is totally plagued (100%)? right? or... if not, what about civil deaths?
            - purge, county level or holding level? I think holding level makes more sense
                - armies can more from holding to holding
                - when all holding in county reach 100%, lich king can get county
        - thw whole questing thing, is that possible and if it is
            - same quests can be get by several?
            - can forces join per quest?
            - for instance, how can I make the healers to reach there but not get obliterared by undeads?
            - I don't like that you have to get quest after quest, maybe when plague is discovered you can get
            a "stop the plague" quest, so you can go anywhere with that excuse
                - but then you can go into Stratholme without doing "the purge"...

    - first you need to create a plague, I checked we are using some sort of artifcat thingy so we could save a "plague blueprint"
        - for that you need a character that can create a plague, like a plague doctor
            - the lich king have some special skills to make those
                - use mind control on place
                    - might result in getting some cultists
                        - cultists can turn into plague doctor
                - raise undead
                    - might get undead champions, can be a plague doctor
        - once you have a player that can create the plague
            - decision to create a plague, chain of events that let's you customize plague
                - this is important, because there are other plagues in the Warcraft lore like the red pox or the hakkar blood
                curse, so we could use this for other stuff in the game
                - chain events let's you customize (in the case of the scourge plague it does a few things things)
                    - extreme health modifier
                    - minor prowess modifier
                    - modifiers like dumb (-1 to attributs)
                    - when it dies it becomes undead
                - you get "blueprint"
            - decision to create quest: go to certain place to spread plague
                - knights like acolytes and necromancer increase chances of going ok
                - knights like rogues help make the mission a success
            - quest can be done by you or you can give it to vassal (use Kel'thuzad as knight or as a ruler to let him go)
                - type of quest: hostile, troops are hidden to everyone but to questgiver and player that does quest, get inside
                enemy territory
                - when you start it it's like a war, you can send troops to a place, but the other doesn't know a "war" has
                started
                - troops are invisible to enemies
                - every time troop move inside someone else terrain they can get discovered
                - quest ends when they reach place
            - when they reach place quest to build up a plague starts
                - territory have 0 to 100 progress for the plague
                - event process (progress goes higher after startup events are done):
                    - startup (0 progress)
                        - explore terrain
                        - prepare plague "factories"
                    - inp progress events
                        - put plague in food
                            - progress goes higher, more chances of being discovered
                        ...
                - decision for plaguin guy
                    - move troops elsewhere to make another plague node
                - event progress for the enemy
                    - some rumors, explore
                    - people get sick
                    - people die
                    - people turns into undead
                        - undead levies and small maa join the plaguing player forces
                - decision for enemy player
                    - explore own terrain, or maybe make another councilor or it's a new task for the marshal
                    - create quest "something is fishy, maybe murlocs?"
                    - create quest "mysterious deaths"
                    - create quest "the dead rising"
                    - create quest "dead land"
                    - create quest "the purge"
                        - player of plagued place can refuse, then a sort of an independence war will pop out
                        - if plagued ruler wins, nothing more happens
                        - else purging player can go to place and start a purge
                            - at this point undead player can throw own troops there, like a war
                                - here, yould send Mal'ganis trops, 
                            - purge takes a time and it's a chain of events too, stuff that can happend
                                - factions event: some think the purge it's too extreme and decide to stop (Jaina, Uther,...)
                                    - you can do a diplomacy check so they can stay
                                    - if you lose, you will lose levies, maa, knights
                                - troops turn into undead
                                ...
                            - at the end of the chain if the purge works then
                    - summary: all these quests is, design a captain, go with small troops there, do something, the purge is special

                - some special stuff
                    - close acolytes might join
                - from 0 to 100 plague, territory produces less and less levies, tax until they become 0
                - at 100 plague
                    - lich king gets control of place
                        - then special war becomes some sort of holy war
                - neighbouring counties also get the plague
    - how the Lordaeron Plague would happen
        - Lich King starts to do the "mind control" on wherever Kel'thuzad is (Dalaran, Scholomance,...? no idea)
        - Lich King gets at least Kel'thuzad as a cultist
            - this means he's secretly a cultist
        - Kel'thuzad moves to Lich King court
        - Lich King teaches Kel'thuzad basic necromancy
        - Lich King posts quest "make a new plague"
        - Lich King has a coutier with plague knowledge, so Lich King uses that guy to complete the quest and start a chained event
            - Lich King picks
                - Extreme health issues
                - low prowess loss
                - low attributs loss
                - raise after undead
            - Lich King gets plague, so quest "make a new plague" is done
        - Lich King posts quest "start plague in Brill"
        - Lich King forms small army with Kel'thuzad as commander and some other kinghts (all acolytes)
        - Brill Plague Brigade (it's a name I'm using here to identify them) starts travelling to Brill, never gets caught
        - they reach place, so quest "start plague in Brill" is done
        - Brill starts being plagued
        
        - from Lich King perspective events pop out
            -  startup of plague
                - explore place
                - make acolyte place deep in the forest
                - invocation of acolytes
                    - basicalle all acolytes close to the place are invited
                    - this results in "voiced" from the forst for the Brill ruler
                - make an small plague lab close to acolyte place
                - start developing plague vials
                - villagers find place, Kel'thuzad decides to test plague on them
                    - we get some undead troops
                - generate enough plague vials
            - plague events
                - wheat is contaminated (increase 20% future increases)
                - after a while plague goes up to 10
                    - the "farms start to lose harvest" for Brill happen
                - Andorhal starts to get plagued too, because startup was done here, no startup is needed there
                - actively making people sick by distributing wheat, making plague up to 30
                    - "people become sick"
                    - "land start to rot"
                - some more villagers find place, Kel'thuzad use plague on them on sew bodies to make an abomination
                - plague to 50
                    - "people die from getting sick", becoming more troops for undead
                - plague goes to 80
                - Kel'thuzad + almost all forces leave Brill to continue stuff on Andorhal

                --- Healer of Quel'talas arrives
                --- Arthas and Jaina arrives, join forces with healers

                - Arthas fights undead forces and win
                - plague goes to 100

        - from Brill ruler perspective
            - event with "voices" heard at night from the darkest places of the forest
            - people dissapear
            - farms start to lose harvest
            
            ----
            - at this point Brill ruler posts quest "something fishy is going on, maybe murlocs?"
            - but nobody pics the quest
            ----

            - people become sick
            - land starts to rot
            - people start dying, not a lot

            ----
            - Brill ruler posts quest "mysterious deaths" and "dead land"
            - some small ruler from Quel'talas picks "dead land" and send an army of healers
            - Arthas picks the "dead land", here is interesting because it should be Arthas + Jaina somehow
                - maybe Jaina is a champion of Arthas court...?
            ----

            - healer from Quel'talas arrives
            - Arthas and Jaina arrives, join forces with healer

            - they both fail their quests

            - Arthas+Jaina+healers get quest from Andorhal and go there
        
        - Andorhal
            - they make the quest with the "the dead rising"

            - almost the same like above, but Kel'thuzad gets killed and before that they moved the whole plague thing to
            Hearthglen too

        - Hearthglen
            - like Andorhal
            
            - here you should get attacked by several forces and at some point Uther helps you with his troops
        - Stratholme
            - Mal'ganis forces joins the main force in Stratholme

            - becomes purged before Arthas comes, but dormant???? how to do this
                - maybe like an special state of plagued
            
            - at this point you can do a "purge" quest, so chain event
                - player controling Stratholme is cool with it
                - Jaina and Uther wanna leave, you try a martial speech and fails, both leave, so you lose troops
                - now Arthas is ready to move troops there, Arthas fight Mal'ganis troops and wins
                - here you can just do a purge and that's it, the whole city is basically destroyed, but it's no longer plagued

**Comment by zumbak04** [comment]
# Answers

> If owning player accepts, then you get the long chain

For every province? It will be spam.

> Cultists from the surrounding areas join (medium increase in success , also minor increase in discovery)

Which cultists? We don't have society.

> Abomination created (medium increase in success, high increase in discovery)

What do you mean?

> Sickened villagers killed by paladins (low decrease in success
> Order of paladins attack (high decrease in success)

What if there are no paladins or Lordaeron is controlled by undead orcs?

> Lordaeron starts purge on Stratholme

You shouldn't take Arthas into the Scourge design. He has its own issue https://github.com/Warcraft-GoA-Development-Team/Warcraft-Guardians-of-Azeroth-2/issues/46.

> Are we gonna have a magic system? with characters being able to cast spells like

It's not done.

> I saw we are using some sort of artifact thingy so we could save a "plague blueprint"

We don't have any artifacts.

> The whole questing thing, is that possible and if it is

You overcomplicate it. That's the Scourge design, what are quests doing here?

> Lich King starts to do the "mind control" on wherever Kel'thuzad is (Dalaran, Scholomance,...? no idea)

What if there's no Kel'Thuzad?

# Comments
- The advanced is overcomplicated. Means you won't finish it in a year. Also, you introduce concepts we don't use (quests) or have so I mostly comment about the simple design.
- A good design doc is supposed to be short and informative. You don't group/split well. You got a giant doc with a lot of indents while you could put every aspect under the title like:
```
# Plague
Here you explain how the Plague works.
Lich King can start it with a decision.
Less taxes, less levies, the Scourge gets spawned troops.

# Purge
Here you explain how the purge works. An event, a decision, I didn't really get what it's.

# Cult
Here you explain what you mean by the cultists.
```
- You introduced the historical design of the Plague (Brill, Andorhal).
  - It's basically "Arthas'" event chain I talked about in #46. So it shouldn't be in this design.
  - Do you take into consideration history may go the other way? Humans lost the Second War, orcs took Lordaeron, etc. So there may be no Brill, there may be no Andorhal, etc.

**Comment by SolTsuki** [comment]
> For every province? It will be spam.
> 
> >
The plague was intended to affect only a county so it's just once. If you want the purge to affect all the counties with plague then it could ask just once.

> Which cultists? We don't have society.
> 
> >
It's just an event, a flavour text, cultists are part of the whole Plague thing so I thought it would made sense. Cultists, or acolytes, won't be a real thing gamewise, just text. 


> What do you mean?
> 
> >
Same thing as the cultists, flavour text.

> What if there are no paladins or Lordaeron is controlled by undead orcs?
> 
> >
Also a flavour text, because paladins only make sense in a Holy Light religion scenario, it should be checked, so it's an specific event for a type of religion. If you have a shamanistic religion you could have the same but with shamans. for instance.


> You shouldn't take Arthas into the Scourge design. He has its own issue #46.
> 
> >
This is an example of how it would happen. Anyone affected by a Plague can do the Purge decision to stop all this. Also, I only mention Arthas at the end, you could use anyone for the "holy war".


> It's not done.

Ok, not needed if we go with the simple design.


> We don't have any artifacts.

Same, unnecessary with simple design.


> You overcomplicate it. That's the Scourge design, what are quests doing here? If you want to use "quests" you have to write their design first, explain what they do, what problem they solve, how they are going to work, how they're visible on the interface, etc.

A quest system would simplify things later, and make them more versatile, and could be used a lot to do other lorewise mechanics. So far we are doing things related to the Warcraft I to III + WoW games, in those games you play as a guy or a group of guys, so what happens in those games is hard to "move" to a game like CK3, which works on a way bigger scale.
That's the quest system for, some Gnolls appeared in the Elwynn Forest? quest to kill the gnolls. As a Gnoll player, you want to get Elwynn forest for yourself? you can still do the Gnoll War, but you can weaken the region first by doing also a quest first.
Think of quests as low scale wars and events, also you can send the quests to allies so they feel a bit like actual quests (a quest should be given, then you complete it). Quests is part of the entire Warcraft universe, so I thought it would add more flavour to it.
In the Scourge of Lordaeron campaign, Arthas is still doing some sort of quests, which is discovering what's going on, then hunting Kel'thuzad, then burning Stratholme down.
Yes, quests woud go first, then the whole Scourge thing. I was pretty sure we would go with the simplified version, so I didn't elaborate more.
Very fast version of the questing system, it would be another tab like the militar one, the council one,... with available quests and ongoing quests, with the description, objectives and reward. Because I saw some artifact files I thought they were implemented, so you could also reward artifacts (weapons, armors, trinkets...) not just gold or lands. The objetives are usually "go there with an army", "kill someone specific", "do a purge decision"


> What if there's no Kel'Thuzad?
You get someone else, in this case you are looking for someone who could work as a necromancer, so any mage could work. Kel'thuzad was a good choice because he already did some experimentation with death magic (here the magic system would make more sense). Stuff like personality traits could be added here, so it's more provable to get a guy who is cruel, a coward,... something like that.


> * The advanced is overcomplicated. Means you won't finish it in a year. Also, you introduce concepts we don't use (quests) or have so I mostly comment about the simple design.

Yep, that's why I did the simplified model. A lot of the stuff like the quests, magic, artificats... take time, but once done you can do more stuff with that. Instead of doing and implementation for every lore mechanic we can start using these building blocks. That's what I had in mind.


> * A good design doc is supposed to be short and informative. You don't group/split well. You got a giant doc with a lot of indents while you could put every aspect under the title like:

Yeah, it could be done way better. I'm not used to do design docs for others. I will redo the simplified version in a moment.

> * It's basically "Arthas'" event chain I talked about in [Third War Storylines #46](https://github.com/Warcraft-GoA-Development-Team/Warcraft-Guardians-of-Azeroth-2/issues/46). So it shouldn't be in this design.

Oh ok, yeah it's in that other task so has nothing to do whith this one.


> * Do you take into consideration history may go the other way? Humans lost the Second War, orcs took Lordaeron, etc. So there may be no Brill, there may be no Andorhal, etc.

Yeah, I presented the design first, then used the Lordaeron's plague as an example of how it would go with each design. You can change places, characters,... the simple version it's just 2 decision + event chains, you can make different flavour texts depending on the situation.

> * I would rather have you focusing on the Plague mechanics only. So no cults, no the Third War, just the Plague.

Ok, I'll redo the simplified version now.

**Comment by SolTsuki** [comment]
# Plague

To start a Plague you need to use the "Start a Plague" decision. To do that you need:
- To select the county you wanna Plague.
- To spend a high quantity of piety (1000?)

Once you do that, an event chain will start, with a 6 months duration, with a success chance and a discovery chance, like in a seduccion scheme.

Some examples of events you would get are:
- Try to recruit cultists (you have diplomacy, intrigue options), if you succeed then success chances increase, but also discovery.
- You try to introduce the plague into the wheat/rice/some culture thing here (intrigue, learning options), if you succeed then success chances increase a lot, discovery increase a bit.

If you get discovered, bad events will start to pop out and the player being plagued gains a casus belli against (not sure about this one, random county under plaguing player, entire realm...) for example:
- Paladins/shamans/religios agent attacks you (intrigue, martial, prowess decision), then success decrease

When the duration is done, a final check is done using the success chances. If the Plague works the county gets plagued, which means:
- Player owning place:
    - Gets no taxes
    - Gets no levies
    - Has no control
    - If they die, plaguing player gets ownership.
- Player that throwed the Plague:
    - Get special troops related to the county size
    - Automatically starts another event chain of the Plague on a surrounding county, with no extra cost.

# Purge

To start a Purge you need to use the "Start a Purge" decision. You need:
- A plagued county, and choose it.
- 250 (?) of piety

2 things can happen at this point, the player owning the plagued county is ok with the Purge or they are not:
- If they are ok with it, the chain event below will start.
- If they are not ok, then a war similar to an independence war will start, if owning player wins the Purge stops, if the purging player wins, the chain event below will start.

Then an event chain starts, duration 2 months, only has success chance, and the events are like:
- You try to get to the plagued county but you find an Ogre camp (Martial, diplomacy and Prowess decisions), if you succeed nothing will happend, there's also the chance of the Ogres joining you, which means more success chances
- Some of your troops are against a Purge (diplomacy, stewarship choices), if you succeed then you get more success chances.
- You find the responsible for the Plague and try to kill them (martial, prowess decisions), if you succeed then you get a high success chance.

If Purge happens, the place is no longer plagued and the purging player earns a casus belli against the plaguin player (same as above, random county or entire realm?)

**Comment by zumbak04** [comment]
> That's the quest system for, some Gnolls appeared in the Elwynn Forest? quest to kill the gnolls. As a Gnoll player, you want to get Elwynn forest for yourself? you can still do the Gnoll War, but you can weaken the region first by doing also a quest first.

That doesn't make much sense if you're Stormwind and have military superiority. Why would Stormwind King travel somewhere and risk their life?

> Quests is part of the entire Warcraft universe, so I thought it would add more flavour to it.

This is a part of the RPG design. It doesn't always look good. I agree Arthas needs it because there's a complicated story with Frostmourne, going to Stratholme, Northrend.

**Comment by zumbak04** [comment]
> like in a seduccion scheme.
 
 It makes sense a lot as a scheme. But I have a question, the ones who joined the scheme, will they become secretly Lich King's followers and get his religion for free?

> Some examples of events you would get are:

You should check default ongoing scheme events, maybe they will fit.
Also, you need bad events too.

> Automatically starts another event chain of the Plague on a surrounding county, with no extra cost.

It may be too spammy. CK3 is about fewer events as possible so you can focus on your realm.

> To start a Purge you need to use the "Start a Purge" decision. You need:
> A plagued county, and choose it.

Again, maybe too spammy. Lordaeron is a lot of counties, it will become a pain to purge them all.

**Comment by SolTsuki** [comment]
> That doesn't make much sense if you're Stormwind and have military superiority. Why would Stormwind King travel somewhere and risk their life?

That's the thing, they won't. The King, or the ruler whose county is attacked, could create the quest, then someone else like another one of the King's vassal could get the quest and complete it for a reward. It's more like a mechanic so players can ask each other for help, while introducing an element constant in the Warcraft universe.


> This is a part of the RPG design. It doesn't always look good. I agree Arthas needs it because there's a complicated story with Frostmourne, going to Stratholme, Northrend.

I agree, sounds good in some cases like the whole Arthas path to Death Knight but it might look bad. Also, this mechanic would require some development time, not sure it's worth it.

**Comment by SolTsuki** [comment]
> It makes sense a lot as a scheme. But I have a question, the ones who joined the scheme, will they become secretly Lich King's followers and get his religion for free?

Didn't think about others joining the scheme, but makes sense, something like "I'll join the Lich King so I can gain his favor later". Also makes sense if they get their religion for free and become secret followers. Would be cool if they could earn "undead status" while doing a pilgrimage.


> You should check default ongoing scheme events, maybe they will fit.
> Also, you need bad events too.

Didn't think about it either, I'll check those. And I'll definetly put some bad events like something with paladins/shamans/whatever fighting against your plague.


> It may be too spammy. CK3 is about fewer events as possible so you can focus on your realm.

This is a tricky one... I really like the feel of plaguing one place, then another and so on until everything is plagued. Maybe the event chain could pop just for the first place and then when it's plagued it can, randomly, every now and then, plague another adjacent place, with no more events. So, if you plague a place, and the plagued player does nothing, it will eventually plague their entire realm.


> Again, maybe too spammy. Lordaeron is a lot of counties, it will become a pain to purge them all.

Maybe with the solution to the above thing this is solved already. You just plague once, plagued counties will randomly plague surrounding counties, with no extra cost and no extra chain events.

**Comment by zumbak04** [comment]
> Didn't think about it either, I'll check those. And I'll definetly put some bad events like something with paladins/shamans/religious unit fighting against your plague.

You may add them now since when the design is over, the coding starts.

**Comment by SolTsuki** [comment]
#V0.0.3

## Plague

To start a Plague you need to use the "Start a Plague" decision. To do that you need:
- To select the county you wanna Plague.
- To spend 1000 piety.

Once you do that, an event chain will start, with a 6 months duration, with a success chance and a discovery chance, like in a seduccion scheme.

Other players might join your scheme, if they do the chances of success increase, also get a minimal increase in discovery.

Some examples of events you would get are:
- Try to recruit cultists (you have diplomacy, intrigue options), if you succeed then success chances increase, but also discovery.
- You try to introduce the plague into the wheat/rice/some culture thing here (intrigue, learning options), if you succeed then success chances increase a lot, discovery increase a bit.
- Someone finds you, they try to escape (intrigue, prowess, diplomacy, martial options), if you succeed then nothing happens (maybe there's an small chance they join you, increasing chances), if you fail, discovery increases a lot.
- You try o create/build <special unit of culture, religion> (an Abomination, a Meatwagon,...) (learning options), if you succeed then success chances increase a lot.
- You try to improve the Plague formula (learning decision), if you succeed then success chances increase a lot, no increase in discvery.
- One of your acolytes try to run away (martial, intrigue, prowess decision), succeeding does nothing, failing means discovery increases a lot.

If you get discovered, bad events will start to pop out, if you fail them then success chances decrease, and the player being plagued gains a casus belli against you (entire realm). for example:
- Paladins/shamans/religios agent attacks you (intrigue, martial, prowess decision)
- Locals tries to attack you (intrigue, martial, prowess, diplomacy, decision)
- Someone manages to sabotage your plague barracks (learning, martial decisions)
- Acolytes get scared and try to run away (diplomacy, intrigue, martial decisions)

When the time is over, a final check is done using the success chances. If the Plague works then:
- Player owning place:
    - Get the county plagued (see below)
- Player that throwed the Plague:
    - Get special troops related to the county size
- Players that joined the plaguing player in the plague:
    - Get the religion of the plaguing player
    - Become secret followers of the plaguing player (see below)

## Plagued county

County effects:
- Gives no taxes
- Gives no levies
- Has no control
- If owning player dies, plaguing player gets ownership
- Every now and then, has a chance of automatically plaguing surrounding counties (with no cost, no chain events attached)

## Purge

To start a Purge you need to use the "Start a Purge" decision. You need:
- A plagued county, and choose it.
- 250 (?) of piety

2 things can happen at this point, the player owning the plagued county is ok with the Purge or they are not:
- If they are ok with it, the chain event below will start.
- If they are not ok, then a war similar to an independence war will start, if owning player wins the Purge stops, if the purging player wins, the chain event below will start.

Then an event chain starts, duration 2 months, only has success chance, and the events are like:
- You try to get to the plagued county but you find an Ogre camp (Martial, diplomacy and Prowess decisions), if you succeed nothing will happend, there's also the chance of the Ogres joining you, which means more success chances
- Some of your troops are against a Purge (diplomacy, stewarship choices), if you succeed then you get more success chances.
- You find the responsible for the Plague and try to kill them (martial, prowess decisions), if you succeed then you get a high success chance.
- Your own men turn intto zombies and try to kill you (martial, prowess, learning decisions), if you succeed nothing happens, if you don't then success chances decrease a lot.
- The try to call the Light/Elements/Religion stuff for help (learning, devotion? decisions), if you succeed then you get a high success chance.
- You try to fight an Abomination/Dreadlord/Lich/religion stuff (martial, prowess decisions), if you succeed then you get a high success chance, if not you get a huge decrease.

If Purge happens, the place is no longer plagued and the purging player earns a casus belli against the plaguin player realm.

## Masters and followers

Appears in the relationship UI, very similar to the lover thing. It's a secret too, so, it can be discovered.
- If a Follower gets discovered, Liege gets a casus belli against him if Master isn't part of hierarchy of Liege. Example: Antonidas discoverers Kel'thuzad is a Follower of Ner'zhul, because Ner'zhul is not related to Antonidas hierarchicaly (isn't a Vassal or a Liege), Antonidas gets a casus belli against Kel'thuzad.
- If a Master gets discovered, well, same as above actually, against the Follower.
- Masters and Followers won't attack each other (AI).
- Later we could add more stuff here like a decision to betray to current Liege to go to the Master's side, but for now this is it.

**Comment by zumbak04** [comment]
> * If a Follower gets discovered, Liege gets a casus belli against him if Master isn't part of hierarchy of Liege. Example: Antonidas discoverers Kel'thuzad is a Follower of Ner'zhul, because Ner'zhul is not related to Antonidas hierarchicaly (isn't a Vassal or a Liege), Antonidas gets a casus belli against Kel'thuzad.

Why do you need a casus belli when there's a crime opinion?

> Your own men turn into zombies and try to kill you (martial, prowess, learning decisions), if you succeed nothing happens, if you don't then success chances decrease a lot.

Your capital may be on another continent, far away from the Plague.

**Comment by zumbak04** [comment]
You can edit your message. No need to copy and resend it.

**Comment by SolTsuki** [comment]
> Why do you need a casus belli when there's a crime opinion?
> 
> >

Hmm.. you are right haha Liege can strip you from titles without incuring tiranny and that's it.


> Your capital may be on another continent, far away from the Plague.

It's just a flavour text, I wasn't thinking about actually reducing your levies, but it would make sense too. Technically even if your capital is on another continent, your are "theoretically" sending your troops to this place, so it makes sense if you lose some even if your capital is on another continent. Actually to make this even more real, I could add a modifier to your levies while purging so you have less, then when the purge is done the modifier also dissapears and you recover levies depending on the event ¿What do you think?

**Comment by zumbak04** [comment]
> It's just a flavour text, I wasn't thinking about actually reducing your levies, but it would make sense too.

When you said you, I thought you meant your own character.

> Your own men turn into zombies and try to kill you

> * If a Follower gets discovered, it's a crime and Liege could use it to strip you of your titles without incurring tyranny. your Master-Follower relationship must be out of the Liege-Vassal hierachy to be a crime. Example: Antonidas discoverers Kel'thuzad is a Follower of Ner'zhul, because Ner'zhul is not related to Antonidas hierarchicaly (isn't a Vassal or a Liege), Kel'thuzad has commited treason.
> * If a Master gets discovered, well, same as above actually, against the Follower.
> * Masters and Followers won't attack each other (AI).
> * Later we could add more stuff here like a decision to betray to current Liege to go to the Master's side, but for now this is it.

I feel like adding the secret faith concept will be more useful in this case and in the future. So when you join the Plague plot, you accept Lich King's faith as your secret.

More useful since it doesn't need a master thus a character. For example, Cho'gall doesn't have a master (since C'Thun won't be added as a character).

If you're going to implement it, check how the False Conversions tenet works. It may be based on this.

**Comment by SolTsuki** [comment]
> I feel like adding the secret faith concept will be more useful in this case and in the future. So when you join the Plague plot, you accept Lich King's faith as your secret.
> 
> More useful since it doesn't need a master thus a character. For example, Cho'gall doesn't have a master (since C'Thun won't be added as a character).
> 
> If you're going to implement it, check how the False Conversions tenet works. It may be based on this.

I like this, it's way better this way (and more simple) and can be used in other cases like the Cho'gall one.

**Comment by SolTsuki** [comment]
# Basic Testing (So Far)
Everything is in the_scourge branch.

If you pick an Undead ruler of Empire level, like Ner'zhul for instance, you can make an hostile character interaction against enemies. It costs 1000 piety though:
![image](https://user-images.githubusercontent.com/86002099/131417407-77248e93-3b19-45cb-9210-3b71c6511622.png)
Then you gotta choose a barony to plague:
![image](https://user-images.githubusercontent.com/86002099/131417446-7fbd4a0b-0d38-4f6f-aefe-f2508388cb70.png)
Then an scheme very similar to a murder one will begin, once it is done you will get a plagued barony like this:
![image](https://user-images.githubusercontent.com/86002099/131417246-92c044ca-8dfe-4da6-bb1e-7e1ed0f15b07.png)
And have some army:
![image](https://user-images.githubusercontent.com/86002099/131417354-b0691537-4130-48e0-9acc-41782b0bcb3b.png)
